xds/internal/resolver/xds_resolver_test.go

```diff
@@ -1459,557 +816,655 @@ func (s) TestResolverMaxStreamDuration(t *testing.T) {
+						RouteSpecifier: &v3httppb.HttpConnectionManager_RouteConfig{
+							RouteConfig: &v3routepb.RouteConfiguration{
+								Name: defaultTestRouteConfigName,
+								VirtualHosts: []*v3routepb.VirtualHost{{
+									Domains: []string{defaultTestServiceName},
+									Routes: []*v3routepb.Route{{
+										Match: &v3routepb.RouteMatch{
+											PathSpecifier: &v3routepb.RouteMatch_Prefix{Prefix: methodName},
+										},
+										Action: &v3routepb.Route_NonForwardingAction{},
+									}},
+								}},
+							}},
+						HttpFilters: []*v3httppb.HttpFilter{e2e.RouterHTTPFilter},
+					}),
 				},
 			},
-			rpcRes: map[string][][]string{
-				"1": {
-					{"build:foo1", "override:foo2", "build:bar1", "override:bar2", "newstream:foo1", "newstream:bar1", "done:bar1", "done:foo1"},
+			wantErr: "matched route does not have a supported route action type",
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			// Spin up an xDS management server.
+			ctx, cancel := context.WithTimeout(context.Background(), defaultTestTimeout)
+			defer cancel()
+			nodeID := uuid.New().String()
+			mgmtServer, _, _ := setupManagementServerForTest(ctx, t, nodeID)
+
+			// Build an xDS resolver.
+			stateCh, _, _ := buildResolverForTarget(t, resolver.Target{URL: *testutils.MustParseURL("xds:///" + defaultTestServiceName)})
+
+			// Update the management server with a listener resource that
+			// contains inline route configuration.
+			configureResourcesOnManagementServer(ctx, t, mgmtServer, nodeID, []*v3listenerpb.Listener{test.listener}, nil)
+
+			// Ensure that the resolver pushes a state update to the channel.
+			cs := verifyUpdateFromResolver(ctx, t, stateCh, "")
+
+			// Ensure that it returns the expected error.
+			_, err := cs.SelectConfig(iresolver.RPCInfo{Method: methodName, Context: ctx})
+			if err == nil || !strings.Contains(err.Error(), test.wantErr) {
+				t.Errorf("SelectConfig(_) = _, %v; want _, Contains(%v)", err, test.wantErr)
+			}
+		})
+	}
+}
+
+func newHTTPFilter(t *testing.T, name, typeURL, path, err string) *v3httppb.HttpFilter {
+	return &v3httppb.HttpFilter{
+		Name: name,
+		ConfigType: &v3httppb.HttpFilter_TypedConfig{
+			TypedConfig: testutils.MarshalAny(t, &v3xdsxdstypepb.TypedStruct{
+				TypeUrl: typeURL,
+				Value: &structpb.Struct{
+					Fields: map[string]*structpb.Value{
+						filterCfgPathFieldName:  {Kind: &structpb.Value_StringValue{StringValue: path}},
+						filterCfgErrorFieldName: {Kind: &structpb.Value_StringValue{StringValue: err}},
+					},
 				},
-			},
-			selectErr: errUnsupportedClientRouteAction.Error(),
+			}),
 		},
+	}
+}
+
+func (s) TestXDSResolverHTTPFilters(t *testing.T) {
+	const methodName1 = "1"
+	const methodName2 = "2"
+	testFilterName := t.Name()
+
+	testCases := []struct {
+		name          string
+		listener      *v3listenerpb.Listener
+		rpcRes        map[string][][]string
+		wantStreamErr string
+		wantPathLen   int
```

Comment:
This field is also never read/asserted on.

Reply:
Oops. Thanks.
As I mentioned earlier, this was one of the hardest tests to read. So, I had tried a few things and clearly forgot to cleanup some that didnt help/work.